format_spec.md

```diff
@@ -0,0 +1,289 @@
+
+---
+
+### `minecraftVersions`
+
+This field MUST be an array listing compatible Minecraft versions for this Project version.
+
+---
+
+### `dependencies` (optional)
+
+This field MUST be an array containing information about dependencies that may or may not be required for this project. Each object in this array contains the following fields:
+
+##### `type`
+
+The type of dependency this is. Valid options are:
+
+- `asset` - For example, a required library mod
+- `framework` - For example, a required modloader
+
+##### `src`
+
+This field contains where this dependency can be downloaded. It contains the name of the host, in an all-lowercase string. Examples include: `curseforge`, `diluv` and `modrinth`. If the `type` field is set to `framework`, this field CANNOT exist;
+
+##### `id`
+
+The ID of the required dependency. If the dependency is a required asset and not a framework, then the ID of the asset stored at the host specified in the `src` field should be used. If the dependency is a required framework, then a framework from the list of standardized framework IDs should be used:
+
+- `fabric-loader`
+- `forge`
+- `liteloader`
+- `rift`
+
+##### `required`
+
+This field MUST be present on dependencies with the `asset` type. It CANNOT be present on dependencies with the `framework` type. This field is a boolean that indicates whether a dependecy is required in order for the Project requiring it to function properly.
+
+##### `version`
+
+This field MUST be either an array or String. If this field is an array, it MUST contain a list of compatible versions. If this field is a String, it MUST contain a Semantic Versioning comparison String. An example of a comparison string is `>= 25.0.219`
+
+---
+
+### `files`
+
+This field contains the downloadable files of a Version. It MUST be an array of File Objects.
+File Objects contain information on a downloadable file. Fields are as follows:
+
+##### `name`
+
+The full name of this file, including file extension.
+
+##### `sha256` (optional)
+
+An SHA256 checksum of the file. This may be used by launchers in order to verify download integiry.
+
+##### `frameworks` (optional)
+
+This field MUST be an array listing Frameworks supported by this file. If no Frameworks are required, this field MAY be omitted or empty. This field MUST contain Strings that use the standard Framework names mentioned in `dependencies.id`
+
+##### `installationMethod` (optional)
+
+The installation method of this file. Valid options are:
+
+- `modsDirectory` - This file must be placed in the `mods` directory
+- `jarMod` - This file is a Jar mod and must be added into minecraft.jar
+- `other`
+
+##### `download`
+
+This field MUST be an array listing available download sources for this file. Each object containing in this array MUST contain the `uri` field, which is a URL pointing to a direct download of the file.
+
```

Comment:
Are source code and deobf files also supported? How are they marked? Rel for files perhaps.